### PR TITLE
Better handling for different callback args

### LINKED ISSUE #9: Lambda callbacks are reported differently than blocks
For example:

``` ruby
class User < ActiveRecord::Base
  before_save do
    # callback.target_symbol will be reported as `block`
  end

  before_save lambda {|u| /* this will be `Proc:0x007fa56d1f7330@somepath.rb` */}
end
```

Perhaps this is intentional, but I thought it was a bit strange to have different output. How do you feel about outputting the following in these scenarios:

``` ruby
class User
  # output the method name `:named_method_callback`
  before_save :named_method_callback

  # output "inline"
  before_save lambda { }
  before_save -> { }
  before_save do
  end

  # output the class name `CallbackClass`
  around_save CallbackClass
end
```

### PR DESCRIPTION
**Pull request by jonmagic**
Closes #9

### CODE REVIEW

**Comment by jch** [comment]
:+1:

File: lib/arca/collector.rb [hunk 1]
```diff
@@ -35,7 +35,18 @@ def self.included(base)
...
             if block
-              args_copy.unshift(:block)
+              args_copy.unshift(:inline)
+            elsif args_copy.first.kind_of?(Proc)
+              args_copy.shift
+              args_copy.unshift(:inline)
+            elsif !args_copy.first.kind_of?(Symbol)
+              klass_or_instance = args_copy.shift
+
+              if klass_or_instance.respond_to?(:new)
```
Comment (jonmagic) [inline lib/arca/collector.rb]: I'm going to go a different direction on this after some thought:

``` ruby
irb(main):010:0> class Foo; end
=> nil
irb(main):011:0> class Bar < Foo; end
=> nil
irb(main):012:0> foo = Foo.new
=> #<Foo:0x007f9f61ac3b38>
irb(main):013:0> bar = Bar.new
=> #<Bar:0x007f9f61abf650>
irb(main):014:0> 
irb(main):015:0* def name_for_object(object)
irb(main):016:1>   if object.class == Class
irb(main):017:2>     object.name
irb(main):018:2>   else
irb(main):019:2*     object.class.name
irb(main):020:2>   end
irb(main):021:1> end
=> :name_for_object
irb(main):022:0> 
irb(main):023:0* name_for_object(Foo)
=> "Foo"
irb(main):024:0> name_for_object(Bar)
=> "Bar"
irb(main):025:0> name_for_object(foo)
=> "Foo"
irb(main):026:0> name_for_object(bar)
=> "Bar"
```

**Comment by jonmagic** [comment]
@jch I just published version 2.2.0 to rubygems.org

File: lib/arca/collector.rb [hunk 2]
```diff
@@ -33,9 +33,21 @@ def self.included(base)
             # Duplicate args before modifying.
             args_copy = args.dup
 
-            # Add target_symbol :block to args_copy if a block was given.
+            # Add target_symbol :inline to args_copy if a block, Proc, or Class
```
Comment (jonmagic) [inline lib/arca/collector.rb:36]: Just realized I got this comment wrong even after updating it :grin: Will fix.

Comment (jch) [inline lib/arca/collector.rb:36]: Heh, oops.

**Pull request merged**